src/openrct2/network/network.cpp

```diff
@@ -534,17 +534,25 @@ const char* Network::FormatChat(NetworkPlayer* fromplayer, const char* text)
 	static char formatted[1024];
 	char* lineCh = formatted;
 	formatted[0] = 0;
+	char timeStamp[128];
+	time_t timer;
+	struct tm * tmInfo;
+	time(&timer);
+	tmInfo = localtime(&timer);
+	strftime(timeStamp, sizeof(timeStamp), "[%H:%M] ", tmInfo);
 	if (fromplayer) {
 		lineCh = utf8_write_codepoint(lineCh, FORMAT_OUTLINE);
 		lineCh = utf8_write_codepoint(lineCh, FORMAT_BABYBLUE);
-		safe_strcpy(lineCh, (const char *) fromplayer->Name.c_str(), sizeof(formatted) - (lineCh - formatted));
+		safe_strcpy(lineCh, timeStamp, sizeof(formatted) - (lineCh-formatted));
+		safe_strcat(lineCh, (const char *) fromplayer->Name.c_str(), sizeof(formatted) - (lineCh - formatted));
```

Comment:
Can you update the buffer length calculations with the value returned from `strftime` (check for 0 too)?